extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosStateDescriptionOptionProvider.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.sonos.internal;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionProvider;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * Dynamic provider of state options while leaving other state description fields as original.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@Component(service = { DynamicStateDescriptionProvider.class,
+        SonosStateDescriptionOptionProvider.class }, immediate = true)
```

Comment:
Why do we need an immediate activation for that component?

Reply:
Is it a valid reason if I tell that all (7) OH2 bindings implementing such provider have set immediate to true ?

Reply:
Not really. :wink: 
It could be set by mistake or perhaps the developer does not understand the instruction and copied it from another component...
Immediate should only be used if the component must be activated also if there is no consumer of that service.
On immediate false (default) the component is activated as soon as someone consumes its but it keeps deactivated as long as this situation is not true.

Reply:
I will make a try without immediate = true.